src/apps/api/tests/test_competitions.py

```diff
@@ -138,3 +142,125 @@ def test_manual_migration_makes_submissions_out_of_only_parents_not_children(sel
+            self.columns = []
+            for columns in range(4):
+                column = ColumnFactory(leaderboard=leaderboard)
+                self.columns.append(column)
+                self.leaderboard_ids_to_columns[leaderboard.id].update({column.title: column.id})
+            for user in self.users:
+                submission = SubmissionFactory(owner=user, phase=self.phase, leaderboard=leaderboard)
+                for col in self.columns:
+                    submission.scores.add(SubmissionScoreFactory(column=col))
+
+    def test_get_competition_leaderboard_as_json(self):
+        # gets makes sure to get JSON response and that it has all leaderboards and users
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        response = c.get(f'/api/competitions/{self.comp.id}/results.json', HTTP_ACCEPT='application/json')
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        self.response_titles = set()
+        self.response_users = set()
+        for key in content.keys():
+            title, id = key.rsplit("(", 1)
+            self.response_titles.add(title)
+            for user in content[key].keys():
+                self.response_users.add(user)
+        leaderboard_titles = set(self.leaderboard_ids_to_titles.values())
+        assert leaderboard_titles == self.response_titles
+        assert self.usernames == self.response_users
+
+    def test_get_competition_leaderboard_by_title_as_json(self):
+        # Makes sure the query returns a json that had a matching leaderboard title
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        leaderboard_choice = random.choice(list(self.leaderboard_ids_to_titles.values()))
+        response = c.get(f'/api/competitions/{self.comp.id}/results.json?title={leaderboard_choice}', HTTP_ACCEPT='application/json')
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        for title in content.keys():
+            assert leaderboard_choice in title
+
+    def test_get_competition_leaderboard_by_id_as_json(self):
+        # Make sure when getting leaderboard by id you get exactly one leaderboard with matching title
+        c = Client()
+        c.login(username="creator2", password="creator2")
+        leaderboard_choice = random.choice(list(self.leaderboard_ids_to_titles.keys()))
+        response = c.get(f'/api/competitions/{self.comp.id}/results.json?id={leaderboard_choice}', HTTP_ACCEPT='application/json')
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+
+        response_title = list(content.keys())
+        assert len(response_title) == 1
+        assert response_title[0] == f'{self.leaderboard_ids_to_titles[leaderboard_choice]}({leaderboard_choice})'
+
+    def test_get_competition_leaderboard_by_id_as_csv(self):
+        c = Client()
```

Comment:
this already exists as `self.client` 

if every test starts with `self.client.login` should probs just put that in `setUp()` and a `logout()` where relevant